.github/workflows/build.yml

```diff
@@ -140,103 +140,122 @@ jobs:
         with:
           fetch-depth: 0
       - name: Build
-        run: docker build --target release -t $DOCKER_IMAGE_ID .
+        run: |
+          docker buildx create \
+            --name multibuilder \
+            --platform linux/amd64,linux/arm64 \
+            --bootstrap --use
+          docker buildx build \
+            --platform linux/amd64,linux/arm64 \
+            -t $DOCKER_IMAGE_ID .
       - name: Check
         run: |
+            docker buildx build --load -t $DOCKER_IMAGE_ID .
+            # Assert that simple cases works for the new built image
             docker run $DOCKER_IMAGE_ID version
             docker run $DOCKER_IMAGE_ID --help
-            docker run $DOCKER_IMAGE_ID help
-            docker run $DOCKER_IMAGE_ID run --help
-            docker run $DOCKER_IMAGE_ID inspect --help
-            docker run $DOCKER_IMAGE_ID status --help
-            docker run $DOCKER_IMAGE_ID stats --help
-            docker run $DOCKER_IMAGE_ID scale --help
-            docker run $DOCKER_IMAGE_ID pause --help
-            docker run $DOCKER_IMAGE_ID resume --help
 
-      - name: Build browser-enabled image
-        run: docker build -t $DOCKER_IMAGE_ID:with-browser --target with-browser .
       - name: Log into ghcr.io
         if: ${{ github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v') }}
         run: |
+          if [[ `git status --porcelain` ]]; then
+            echo "The repository has unexpected changes identified from the git history."\
+            "Force the fail as the job can't guarantee that the code is not changed between the previous build and the next build+push."
+            exit 1
+          fi
```

Comment:
does `docker buildx build` touches local directory?

This seems strange. I am not against us checking, just seems strange for us to need it given that there is nothing really that we do that is expected to touch local directory here.

Reply:
I removed, this was mostly a request by Andrey. But I think we can remove and add in the future when he is back.